exchanges/order/orders.go

```diff
@@ -676,3 +690,83 @@ func (o *ClassificationError) Error() string {
 		o.Exchange,
 		o.Err)
 }
+
+// StandardCancel defines an option in the validator to make sure an ID is set
+// for a standard cancel
+func (c *Cancel) StandardCancel() validate.Checker {
+	return validate.Check(func() error {
+		if c.ID == "" {
+			return errors.New("ID not set")
+		}
+		return nil
+	})
+}
+
+// Validate checks internal struct requirements
+func (c *Cancel) Validate(opt ...validate.Checker) error {
+	if c == nil {
+		return ErrCancelOrderIsNil
+	}
+
+	if c.Pair.IsEmpty() {
+		return errors.New("currency pair not set")
+	}
+
+	if c.AssetType.String() == "" {
+		return errors.New("asset type not set")
+	}
+
+	var errs common.Errors
+	for _, o := range opt {
+		err := o.Check()
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if errs != nil {
+		return errs
+	}
+	return nil
+}
+
+// Validate checks internal struct requirements
+func (g *GetOrdersRequest) Validate(opt ...validate.Checker) error {
+	if g == nil {
+		return ErrGetOrdersRequestIsNil
+	}
+	var errs common.Errors
+	for _, o := range opt {
+		err := o.Check()
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if errs != nil {
+		return errs
+	}
+	return nil
+}
+
+// Validate checks internal struct requirements
+func (m *Modify) Validate(opt ...validate.Checker) error {
```

Comment:
So it didn't post my nit.... :D

I agree with @MadCozBadd , I think the asset type check should be extended to `modify` and `submit` validations as we have wrapper code which will change behavious based on asset type